src/main/java/com/exadel/discountwebapp/discount/service/DiscountService.java

```diff
@@ -43,9 +45,17 @@
     public DiscountResponseVO findById(Long id) {
         Discount discount = discountRepository.findById(id)
                 .orElseThrow(() -> new EntityNotFoundException(Discount.class, "id", id));
-        Long quantity = discount.getViewNumber() == null ? 1 : discount.getViewNumber() + 1;
-        discount.setViewNumber(quantity);
-        discountRepository.save(discount);
+        while (true) {
+            try {
+                Long quantity = discount.getViewNumber() == null ? 1 : discount.getViewNumber() + 1;
+                discount.setViewNumber(quantity);
+                discount.setVersion(discount.getVersion());
```

Comment:
we shouldn't manage version field by ourselves, this field should be managed by Hibernate